clients/drcachesim/scheduler/scheduler.h

```diff
@@ -623,6 +625,15 @@ template <typename RecordType, typename ReaderType> class scheduler_tmpl_t {
          * sensitivity studies.
          */
         bool randomize_next_input = false;
+        /**
+         * If true, the scheduler will read from each input to determine its filetype
+         * during initialization.  If false, the filetype will not be available prior
+         * to explicit record rerieval by the user, but this may be required for
+         * inputs whose sources are not yet set up at scheduler init time.
```

Comment:
I was wondering in what case this must be set to false. Can you give an example of "inputs whose sources are not yet set up at scheduler init time"?

Reply:
Online analysis sets up IPC readers which open pipes and those pipes block.  The launcher wants to initialize the analyzer+scheduler and then go run the app which will unblock the pipes with data so we want init routines to not block.  Added the example to this comment.